### PR TITLE
Add support to Synthea so it can now either generated 100 alive patie…

### PR DESCRIPTION
**Pull request by AmartC**
…nts or 100 dead patients. In addition, ensure that causes of death are always based on modifying some of the modules

### CODE REVIEW

**Comment by dehall** [comment]
The `run_random_dead` function is very similar to the `run_random` function so it might make more sense to combine those and have the check on `Synthea::Config.sequential.all_dead` be internal to that combined function, so that if we want to make changes in the future we only have to do it in one spot.

Also, the config setting only applies to running a random simulation, not the simulation based on targeted city populations (ex `bundle exec rake synthea:generate['./config/Suffolk_County.json'] ` to generate patients in that one county, with realistic but scaled populations and demographics ) so I think the setting name should indicate that somehow.

**Comment by AmartC** [comment]
Yeah, I had them as one function but there were a lot of conditionals I added and it seemed a bit messier so I decided to make it as two functions. I can go back and change the variable names since the focus was to have Synthea generate the patients one by one rather than through city information.

File: config/synthea.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ synthea:
     generate_fingerprints: false
   sequential: # sequential population generation... one at a time...
     # population is scaled to accurately represent MA (using a "true" population of 6,794,422)
+    random_all_dead: false
```
Comment (jawalonoski) [inline config/synthea.yml]: Maybe `only_dead_patients` is a better description.

File: lib/world/sequential.rb [hunk 2]
```diff
@@ -45,6 +45,7 @@ def initialize(datafile = nil)
         }
 
         @population_count = Synthea::Config.sequential.population
+        @random_all_dead_bool = Synthea::Config.sequential.random_all_dead
```
Comment (jawalonoski) [inline lib/world/sequential.rb]: Again, something like `@only_dead_patients` is more descriptive/readable.

**Comment by jawalonoski** [comment]
Some variable name changes and we're ready to go here.

**Pull request merged**